### PR TITLE
[Functionalization] Remove View in tensor_methods::transpose

### PR DESCRIPTION
**Pull request by wonjoolee95**
[Functionalization] Remove View in tensor_methods::transpose

---

Test plan:
- Passing locally: 
   - `PJRT_DEVICE=CPU python ../test/test_view_ops.py -v -k TestViewOpsXLA.test_transpose_view`
   - `PJRT_DEVICE=CPU python test/test_operations.py TestAtenXlaTensor.test_transpose_1d`
   - `PJRT_DEVICE=CPU python test/test_operations.py TestAtenXlaTensor.test_transpose_1d_inplace`

### CODE REVIEW

**Review by alanwaketan — APPROVED** [review]
LGTM.

File: torch_xla/csrc/tensor_methods.cpp [hunk 1]
```diff
@@ -2640,18 +2640,36 @@ XLATensorPtr trace(const XLATensorPtr& input) {
...
-                         GetXlaShape(ir_value));
+    // kNoOp
+    result = input;
   } else {
+    // kPermute
     std::vector<int64_t> permute_dims = torch::lazy::MakeTransposePermutation(
         /*dim0=*/dim0, /*dim1=*/dim1, /*rank=*/input_shape.get().rank());
-    view_info = ViewInfo(ViewInfo::Type::kPermute, input_shape, permute_dims);
+    result = input->CreateFrom(torch::lazy::MakeNode<Permute>(
+        input->GetIrValue(), xla::InversePermutation(permute_dims)));
```
Comment (alanwaketan) [inline torch_xla/csrc/tensor_methods.cpp:2669]: Wondering why we need InversePermutation?

Comment (vanbasten23) [reply torch_xla/csrc/tensor_methods.cpp:2669]: maybe here? https://github.com/pytorch/xla/blob/1a4e43280181231c60e82e522caf3c0882531e1e/torch_xla/csrc/view.cpp#L93-L95

**Review by vanbasten23 — APPROVED** [review]
LGTM. Thanks.

File: torch_xla/csrc/tensor_methods.cpp (hunk 1 above)
Comment (wonjoolee95) [reply torch_xla/csrc/tensor_methods.cpp:2669]: Oh good catch, this should be actually without `xla:: InversePermutation`. I fixed it in the other PR for permute (https://github.com/pytorch/xla/pull/5195/commits/e37a9863d15cb99f764c797cd18c8f44ff9d4ba0), but must have missed it here. Let me update it. Thanks!

**Pull request merged**